### PR TITLE
feat(contracts): add two steps ownership transfer

### PR DESCRIPTION
**Pull request by PierrickGT**

### CODE REVIEW

File: contracts/OwnerOrManager.sol [hunk 1]
```diff
@@ -1,72 +1,112 @@
...
-     * @dev Throws if called by any account other than the owner.
      * @param _newManager New _manager address.
      * @return Boolean to indicate if the operation was successful or not.
      */
     function _setManager(address _newManager) internal returns (bool) {
         address _previousManager = _manager;
-        require(_newManager != address(0), "Manager/manager-not-zero-address");
-        require(_newManager != _previousManager, "Manager/existing-manager-address");
+
+        require(_newManager != address(0), "OwnerOrManager/manager-not-zero-address");
```
Comment (asselstine) [inline contracts/OwnerOrManager.sol]: They should be able to clear the manager, I think setting to address(0) should be allowed

File: contracts/OwnerOrManager.sol [hunk 2]
```diff
@@ -1,72 +1,112 @@
...
         _;
     }
 
+    /**
+    * @dev Allows current owner to set the pendingOwner address.
+    * @param _newOwner Address to transfer ownership to.
+    */
+    function transferOwnership(address _newOwner) public override onlyOwner {
+        require(_newOwner != address(0), "OwnerOrManager/pendingOwner-not-zero-address");
+        _pendingOwner = _newOwner;
```
Comment (asselstine) [inline contracts/OwnerOrManager.sol]: Emit a new event here.

OwnershipOffered?

**Review by asselstine — CHANGES_REQUESTED** [review]
Move all 'pending' ownership logic into a new "Ownable" contract that is entirely ours

File: contracts/OwnerOrManager.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/OwnerOrManager.sol]: Done: https://github.com/pooltogether/owner-manager-contracts/pull/2/commits/3a12659166cf1b2bf688be81d760cb1a76c6a045

File: contracts/OwnerOrManager.sol (hunk 2 above)
Comment (PierrickGT) [reply contracts/OwnerOrManager.sol]: Done: https://github.com/pooltogether/owner-manager-contracts/pull/2/commits/bbeb195c9aed908f212916c35caf0b130d97118a

**Review by asselstine — CHANGES_REQUESTED** [review]
"Ownable" needs to be it's own separate contract.

"Manageable" can then inherit from Ownable, and add manager stuff.

(That's what I meant when I said copy in OZ and make our own Ownable)

File: contracts/Ownable.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.0;
+
+import "./external/openzeppelin/Context.sol";
```
Comment (asselstine) [inline contracts/Ownable.sol]: ✂️  we don't need this

File: contracts/Ownable.sol [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+     * @param newOwner new `_owner` address.
+     */
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    /* ============ Deploy ============ */
+
+    /**
+     * @notice Initializes the contract setting the deployer as the initial owner.
+     */
+    constructor() {
```
Comment (asselstine) [inline contracts/Ownable.sol]: Pass the owner to the constructor, otherwise deploying will be hard

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/Ownable.sol (hunk 3 above)
Comment (PierrickGT) [reply contracts/Ownable.sol]: Done: https://github.com/pooltogether/owner-manager-contracts/pull/2/commits/bc63dde7817d9fba3f8b8d8344a61bdaa1a728ba

File: contracts/Ownable.sol (hunk 4 above)
Comment (PierrickGT) [reply contracts/Ownable.sol]: Done: https://github.com/pooltogether/owner-manager-contracts/pull/2/commits/c29bd611dc008202d7ecb499d26972256bc00179

**Review by asselstine — APPROVED** [review]

**Comment by asselstine** [comment]
Just tests

**Pull request merged**